core/utils/mailbox_prom.go

```diff
@@ -55,15 +61,21 @@ func (m *MailboxMonitor) HealthReport() map[string]error {
 	return map[string]error{m.Name(): m.StartStopOnce.Healthy()}
 }
 
-func (m *MailboxMonitor) monitorLoop(c <-chan time.Time) {
-	for range c {
-		m.mailboxes.Range(func(k, v any) bool {
-			name, mb := k.(string), v.(mailbox)
-			c, p := mb.load()
-			capacity := strconv.FormatUint(c, 10)
-			mailboxLoad.WithLabelValues(m.appID, name, capacity).Set(p)
-			return true
-		})
+func (m *MailboxMonitor) monitorLoop(ctx context.Context, c <-chan time.Time) {
+	defer close(m.done)
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
In the case the context is canceled, should we look for any messages left in c?

Reply:
Leftover messages aren't significant in this case because we are just reporting a metric, so returning promptly is more important.

If the work was significant and could not be dropped, then the old approach of consuming the channel until it is closed would be more appropriate (assuming it were properly closed) since it directly expresses that. I try to avoid [Context](https://pkg.go.dev/context) in cases like that where we are just signalling, and don't intend to respect the prescribed usage:
> Deadline returns the time when work done on behalf of this context should be canceled.
>
> A CancelFunc tells an operation to abandon its work.
>
> Canceling this context releases resources associated with it...